### PR TITLE
added test_asn_info.py

### PR DESCRIPTION
**Pull request by Sekhar-Kumar-Dash**
## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Review by AlyaGomaa — CHANGES_REQUESTED** [review]
Hey @Sekhar-Kumar-Dash thanks for your effort, i left you a few comments

File: tests/test_asn_info.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+            {"asn": {"number": "AS13335", "org": "CLOUDFLARENET, US"}},
+        ),
+        # Testcase 3: Cache hit, return cached data
+        ("8.8.8.8", {"asn": {"number": "AS15169", "org": "GOOGLE, US"}}),
+    ],
+)
+def test_cache_ip_range(mock_db, ip_address, expected_cached_data):
+    ASN_info = ModuleFactory().create_asn_obj(mock_db)
+    ASN_info.cache_ip_range(ip_address)
+    assert ASN_info.cache_ip_range(ip_address) == expected_cached_data
```
[the remark this answers is not in the document]
Comment (AlyaGomaa) [reply tests/test_asn_info.py]: @Sekhar-Kumar-Dash this one isn't done yet, right?

**Review by AlyaGomaa — CHANGES_REQUESTED** [review]
hi there's 1 comment that you forgot to resolve @Sekhar-Kumar-Dash

**Comment by Sekhar-Kumar-Dash** [comment]
> hi there's 1 comment that you forgot to resolve @Sekhar-Kumar-Dash

@AlyaGomaa  i completely missed that commnet i have fixed it now please take a look

**Comment by AlyaGomaa** [comment]
yes awesome thank youu! @Sekhar-Kumar-Dash

**Pull request merged**